frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ProcessExecutor.java

```diff
@@ -67,7 +73,10 @@ public int execute(final Logger logger, final OutputStream stdout, final OutputS
             throws ProcessExecutionException {
         logger.debug("Executing command line {}", commandLine);
         try {
-            ExecuteStreamHandler streamHandler = new PumpStreamHandler(stdout, stderr);
+            // This will normally include the current maven module in multithreaded builds
+            // Kept deliberately short as Maven logs are already very wide
+            String executorName = Thread.currentThread().getName() + "-FE";
+            ExecuteStreamHandler streamHandler = new NamedPumpStreamHandler(stdout, stderr, executorName);
```

Comment:
I thought about instead about including the specific execution ID, but this requires drilling through a couple of dozen classes and that would make keeping in sync with upstream a lot harder.